SwiftCompilerSources/Sources/Optimizer/Utilities/ManglingUtils.swift

```diff
@@ -0,0 +1,129 @@
+//===--- ManglingUtils.swift --------------------------------------------==//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SIL
+import OptimizerBridging
+
+/// Represents a parse tree for a demangled string.
+/// 
+/// The type is not directly constructible and can only be obtained 
+/// by demangling a mangled string using the `Demangler`. The lifetime of
+/// such a `DemangledNode` ends with the lifetime of the `Demangler`.
+struct DemangledNode {
```

Comment:
It's fine to bridge all the demangler internals, but it's not really needed for the closure specializer pass. We can add a single bridging API to mangle the closure specialized functions, like done here: https://github.com/apple/swift/blob/d8b5b889b330beba5a1920766ebbfce0f4662cf5/SwiftCompilerSources/Sources/Optimizer/PassManager/ModulePassContext.swift#L158